src/org/pentaho/reporting/platform/plugin/async/IAsyncReportListener.java

```diff
@@ -2,7 +2,7 @@
 
 import org.pentaho.reporting.engine.classic.core.event.ReportProgressListener;
 
-public interface IAsyncReportListener extends ReportProgressListener {
+public interface IAsyncReportListener extends ReportProgressListener, IAsyncReportState {
 
```

Comment:
JobManager uses IAsyncReportState as a read-only state and that is the main reason of implementing such interface. The another - having this interface may (or may not) help to have ability to serialize report status - taking a bit close look at JobManager. So:

- IAsyncReportState  is read only interface
- We serialize report state (see JobManager.getState()) to json (possibly for xml) according to THIS interface
- we don't mess up 'read only serialization' with 'IAsyncReportListener' methods that do expose listener as a listener

Do we still need to get rid of IAsyncReportState?

"PentahoAsyncReportExecution" does not know it's own id if it has no listener. InstanceId - is ID nested from BackgroundJobReportContentGenerator. Remember task is:

1. queued
2. working
3. finished

so this states will be:

1. BackgroundJobReportContentGenerator->PentahoAsyncExecutor
2. PentahoAsyncExecutor
3. PentahoAsyncExecutor -> JobManager (retuns content)

I want to track a gap how long a report execution action lives between creation of callable and actual execution. This could be tracked (I assume could) with audit records with same instance ID: BackgroundJobReportContentGenerator creates callable and makes audit record with it's id. Then - callable has an instance id of BackgroundJobReportContentGenerator - it make record that execuiotn actually started. Listener UUID is different. Listener ID is UUID used by JobManager to track execution status. So instanceId - is for audit purposes, listener id - to determine execuiton id which is different. I am not sure I can use ContentGenerator ID as ID for async job - so we have 2 separate ids now. Listener do not share instanceId with PentahoAsyncReportExecution.

Is it sounds reasonable?